okio/src/commonMain/kotlin/okio/OkioMessageDigestAlgorithm.kt

```diff
@@ -0,0 +1,8 @@
+package okio
+
+enum class OkioMessageDigestAlgorithm(val value: String) {
```

Comment:
I would drop this. While the type safety is nice, we only expose the available options as functions rather than accept it as a parameter in the public API. We can just use strings internally and save the class.

Reply:
This scratches the surface of something I'm slightly concerned about. 

Because were separating modules of `commonMain`, `commonTest`, `nonJvmMain`, etc. We can't directly test `OkioMessageDigest` if its marked internal. 

Currently, `newMessageDigest` and `OkioMessageDigest` are public. I don't know if a better approach is to test this message digest by proxy via `ByteString` and `HashingSink`. 

Thoughts?

Reply:
Internal visibility should be available from tests. The IDE probably just isn't smart enough, but it should compile from the command line.

With Kotlin 1.4 the idea of a non-JVM source set will get more first-class support.

Reply:
Oh but yeah to answer your question, using the public API from commonTest is far-preferred. We want to use the same codepaths as our users, not rely on implementation details.

Reply:
Okay, I'll give this a go tomorrow and see how it plays out. Thanks 🙂

Reply:
> Oh but yeah to answer your question, using the public API from commonTest is far-preferred. We want to use the same codepaths as our users, not rely on implementation details.

To clarify, meaning the tests backing this should go through ByteString?

Reply:
Yep!